### ISSUE TITLE
Credo.Check.Readability.StringSigils parsing error

### ISSUE THREAD

**Issue by VitorTrin** [rrrene/credo#695]
### Environment

* Credo version (`mix credo -v`): 1.1.3
* Erlang/Elixir version (`elixir -v`): 1.8.2 OTP 21
* Operating system: Manjaro

### What were you trying to do?

```elixir
function.("'#{parameter}'")
```

### Expected outcome

No errors, no warnings

### Actual outcome

```
warning: Unexpected error while parsing lib/credo_crash.ex: [%Credo.Issue{category: :error, check: Credo.Code.ParserError, column: nil, exit_status: 0, filename: "nofilename", line_no: 18, message: "unexpected token: ", meta: [], priority: 0, scope: nil, severity: nil, trigger: nil}]
  lib/credo/check/readability/string_sigils.ex:50: Credo.Check.Readability.StringSigils.run/2
  lib/credo/check/runner.ex:168: Credo.Check.Runner.do_run_check_on_single_source_file/3
  lib/credo/check/runner.ex:161: Credo.Check.Runner.run_check_on_single_source_file/3
  (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
  (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
```

Please note, if I do

```elixir
function.(
  "'#{parameter}'"
  )
```

there are no crashes

**Comment by rrrene** [comment]
Thx for reporting this issue :+1: 

I tried to reproduce this but failed. You can see what I tried here: d757fdb

Can you provide any additional information that could help track down this error?

**Comment by VitorTrin** [comment]
Hi, just came back from vacation.
I forgot to mention the problem requires credo to be run with `--strict`
Here is the code I used to reproduce https://github.com/VitorTrin/credo_crash

**Comment by ityonemo** [comment]
does not crash when running --strict with:

credo:
1.1.3

elixir:
Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
Elixir 1.9.1 (compiled with Erlang/OTP 21)

**Comment by TheFirstAvenger** [comment]
@rrrene I was able to reproduce this via the example repo given and running Elixir `1.8.2`. Looks like `Heredocs.replace_with_spaces()` is turning:
`"defmodule CredoCrash do\n  @moduledoc false\n  def hello(function) do\n    parameter = 1\n\n    function.(\"'\#{parameter}'\")\n  end\nend\n"`
into the invalid code:
`"defmodule CredoCrash do\n  @moduledoc false\n  def hello(function) do\n    parameter = 1\n\n    function.(\"'\#{            )\n  end\nend\n"`

in Elixir `1.8.2` (notice the mishandling of `#{`)

It instead turns it into:
`"defmodule CredoCrash do\n  @moduledoc false\n  def hello(function) do\n    parameter = 1\n\n    function.(\"'            '\")\n  end\nend\n"`

in `1.9.4`, which is valid and why this issue doesn't happen on `1.9.4`.

**Comment by TheFirstAvenger** [comment]
I have confirmed that the root issue here is a defect in `:elixir_tokenizer` in 1.8.x that was resolved in 1.9.0. This is the PR in elixir that fixed the bad token handling: https://github.com/elixir-lang/elixir/pull/8995.

I don't believe there is anything we can do in Credo to resolve this, so the end user resolution would be to either upgrade to Elixir 1.9.0+ or disable the StringSigils check. I recommend closing this issue.
